janusgraph-backend-testutils/src/main/java/org/janusgraph/graphdb/JanusGraphIndexTest.java

```diff
@@ -1523,6 +1529,202 @@ public void testIndexParameters() {
+        }};
+        checkMixedIndexCountProfiling(profile, annotations);
+        assertEquals(total, graph.traversal().V().has("name", P.neq("value")).count().next());
+
+        profile = graph.traversal().V().has("name", P.neq(null)).count().profile().next();
+        annotations = new HashMap() {{
+            put("query", String.format("[(%s <> null)]:vsearch", nameKey));
+        }};
+        checkMixedIndexCountProfiling(profile, annotations);
+        assertEquals(total, graph.traversal().V().has("name", P.neq(null)).count().next());
+
+        /* ==========================================================================
+        has(key) followed by count() can be optimized */
+
+        total = numV;
+        profile = graph.traversal().V().has("name").count().profile().next();
+        annotations = new HashMap() {{
+            put("query", String.format("[(%s <> null)]:vsearch", nameKey));
+        }};
+        checkMixedIndexCountProfiling(profile, annotations);
+        assertEquals(total, graph.traversal().V().has("name").count().next());
+
+        /* ==========================================================================
+        has(key, value) followed by other steps and then count() cannot be optimized */
+
+        total = numV / strings.length;
+        profile = graph.traversal().V().has("name", "Uncle Berry has a farm").out().count().profile().next();
+        assertFalse(profile.getMetrics(0).getName().contains(JanusGraphMixedIndexCountStep.class.getSimpleName()));
+        assertTrue(profile.getMetrics(0).getName().contains(JanusGraphStep.class.getSimpleName()));
+        assertEquals(total, graph.traversal().V().has("name", "Uncle Berry has a farm").out().count().next());
+
+        /* ==========================================================================
+        has(key, value) with composite index followed by count() cannot be optimized */
+
+        final PropertyKey ageProperty = mgmt.makePropertyKey("age").dataType(Integer.class).make();
+        mgmt.buildIndex("ageidx", Vertex.class).addKey(ageProperty).buildCompositeIndex();
+        finishSchema();
+
+        for (int i = 0; i < 10; i++) {
+            tx.addVertex("age", i % 2);
+        }
+        newTx();
+
+        profile = graph.traversal().V().has("age", 0).count().profile().next();
+        assertFalse(profile.getMetrics(0).getName().contains(JanusGraphMixedIndexCountStep.class.getSimpleName()));
+        assertTrue(profile.getMetrics(0).getName().contains(JanusGraphStep.class.getSimpleName()));
+        assertEquals(5, graph.traversal().V().has("age", 0).count().next());
+        profile = graph.traversal().V().has("age").count().profile().next();
+        assertFalse(profile.getMetrics(0).getName().contains(JanusGraphMixedIndexCountStep.class.getSimpleName()));
+        assertTrue(profile.getMetrics(0).getName().contains(JanusGraphStep.class.getSimpleName()));
+        assertEquals(10, graph.traversal().V().has("age").count().next());
+
```

Comment:
I was thinking about more test cases. Can we add 2 more test cases?
`has("nonIndexedProperty").has("indexedProperty").count()`
`has("indexedProperty").has("nonIndexedProperty").count()`
What do you think about it? Maybe we already checking these cases in other tests. If so, just ignore this comment 
the above 2 queries are computed the same but they should not be optimized with Mixed Query Count optimization because the whole chain of 'has' steps cannot be answered by a single mixed index.